app/assets/javascripts/3dbio_viewer/src/webapp/components/model-upload/ModelUpload.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useRef } from "react";
+import { Dialog, DialogContent, DialogTitle, IconButton } from "@material-ui/core";
+import { Close } from "@material-ui/icons";
+import _ from "lodash";
+import i18n from "../../utils/i18n";
+import { Dropzone, DropzoneRef } from "../dropzone/Dropzone";
+import { useAppContext } from "../AppContext";
+import "./ModelUpload.css";
+
+export interface ModelUploadProps {
+    title: string;
+    onClose(): void;
+}
+
+export const ModelUpload: React.FC<ModelUploadProps> = React.memo(props => {
+    const { title, onClose } = props;
+    const { compositionRoot } = useAppContext();
+
+    const [jobTitle, setJobTitle] = useState<string>("");
+    const structureFileRef = useRef<DropzoneRef>(null);
+    const annotationFileRef = useRef<DropzoneRef>(null);
+    const [error, setError] = useState<string>();
+
+    const onSubmitHandler = () => {
```

Comment:
Tipically, this would use `React.useCallback`. Here, it's a callback which is performing a side effect. We have a custom hook for that: `useCallbackEffect`. See `ModelSearch.tsx` for an example. It's a callback + cancellable effect